api/scpca_portal/views/dataset.py

```diff
@@ -49,3 +49,26 @@ def destroy(self, request, pk=None):
             {"detail": "Deleting datasets is not allowed."},
             status=status.HTTP_405_METHOD_NOT_ALLOWED,
         )
+
+    def perform_create(self, serializer):
+        dataset = serializer.save()
+        if dataset.start:
+            self.submit_job(dataset)
+
+    def perform_update(self, serializer):
+        dataset = serializer.save()
+        if dataset.start:
+            # If dataset already has active job, don't allow user to spawn additional job
+            if dataset.is_started or dataset.is_processing:
+                return
```

Comment:
This should return an error that you cannot modify a dataset that is currently being processed.